system_metrics_collector/include/moving_average_statistics/types.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MOVING_AVERAGE_STATISTICS__TYPES_HPP_
+#define MOVING_AVERAGE_STATISTICS__TYPES_HPP_
+
+/**
+ *  A container for statistics results
+ */
+struct StatisticResults
+{
+  double average;
+  double min;
+  double max;
+  double standard_deviation;
+  int sample_count;
```

Comment:
You can probably increase `sample_count` to `long` without increasing `sizeof(StatisticResults)` due to padding.

Reply:
This actually needs to be int64_t in order to match the observed value.